src/userscriptMetadata.js

```diff
@@ -54,6 +55,8 @@ export async function generateMetadataBlock(userscriptPath) {
 		...await loadMetadata(userscriptPath),
 	};
 
+	metadata.homepageURL = gitRepo.readmeSectionUrl({ baseName, metadata });
```

Comment:
This doesn't feel quite right, but we have a circular dependency - we need an instance of `UserscriptDefaultMetadata` in order to generate the `homepageURL` property.

Reply:
Yeah, but I don't see a way around this. I think it should be fine as long as nobody expects to be able to construct the section name using the homepage URL. Even in that case we have a sensible default as the empty URL fragment will still be a valid link to the repo.